swarm/src/behaviour.rs

```diff
@@ -299,21 +290,20 @@ pub enum ToSwarm<TOutEvent, TInEvent> {
         event: TInEvent,
     },
 
-    /// Informs the `Swarm` about an address observed by a remote for
-    /// the local node by which the local node is supposedly publicly
-    /// reachable.
+    /// Reports a new candidate for an external address to the [`Swarm`].
     ///
-    /// It is advisable to issue `ReportObservedAddr` actions at a fixed frequency
-    /// per node. This way address information will be more accurate over time
-    /// and individual outliers carry less weight.
-    ReportObservedAddr {
-        /// The observed address of the local node.
-        address: Multiaddr,
-        /// The score to associate with this observation, i.e.
-        /// an indicator for the trusworthiness of this address
-        /// relative to other observed addresses.
-        score: AddressScore,
-    },
+    /// This address could come from a variety of sources:
+    /// - A protocol such as identify obtained it from a remote.
+    /// - The user provided it based on configuration.
+    /// - We made an educated guess based on one of our listen addresses.
```

Comment:
```suggestion
```

I would expect one to use the `TransportEvent::NewListener` event. In what use-case would a `NetworkBehaviour` then have the need to also emit a `ToSwarm::NewExternalAddrCandidate`?

Reply:
Who is "one"?

There is only so much we can assume about the transport stack within `Swarm`. Users might know more about the network environment they are running in so they can write a `NetworkBehaviour` that based on incoming listen addresses knows who to generate candidates for external addresses. That is what I meant with "educated guesses".